nav2_costmap_2d/src/costmap_2d_ros.cpp

```diff
@@ -875,6 +876,7 @@ void Costmap2DROS::getCostsCallback(
       response->costs.push_back(static_cast<float>(costmap->getCost(mx, my)));
     }
   }
+  lock.unlock();
```

Comment:
I don't think this is explicitly necessary. When the lock goes out of scope it unlocks